test/e2e/network/netpol/test_helper.go

```diff
@@ -0,0 +1,161 @@
+		framework.Logf("Network Policy creating %s/%s \n%s", namespace, policy.Name, prettyPrint(policy))
+		framework.Logf("****************************************************************")
+	}
+
+	_, err := k8s.CreateNetworkPolicy(namespace, policy)
+	framework.ExpectNoError(err, "Unable to create netpol %s/%s", namespace, policy.Name)
+}
+
+// UpdatePolicy updates a networkpolicy
+func UpdatePolicy(k8s *Scenario, policy *networkingv1.NetworkPolicy, namespace string) {
+	if isVerbose {
+		framework.Logf("****************************************************************")
+		framework.Logf("Network Policy updating %s/%s \n%s", namespace, policy.Name, prettyPrint(policy))
+		framework.Logf("****************************************************************")
+	}
+
+	_, err := k8s.UpdateNetworkPolicy(namespace, policy)
+	framework.ExpectNoError(err, "Unable to update netpol %s/%s", namespace, policy.Name)
+}
+
+// ValidateOrFail validates connectivity
+func ValidateOrFail(k8s *Scenario, model *Model, testCase *TestCase) {
+	ginkgo.By("Validating reachability matrix...")
+
+	// 1st try
+	ginkgo.By("Validating reachability matrix... (FIRST TRY)")
+	ProbePodToPodConnectivity(k8s, model, testCase)
+	// 2nd try, in case first one failed
+	if _, wrong, _, _ := testCase.Reachability.Summary(ignoreLoopback); wrong != 0 {
+		framework.Logf("failed first probe %d wrong results ... retrying (SECOND TRY)", wrong)
+		ProbePodToPodConnectivity(k8s, model, testCase)
+	}
+
+	// at this point we know if we passed or failed, print final matrix and pass/fail the test.
+	if _, wrong, _, _ := testCase.Reachability.Summary(ignoreLoopback); wrong != 0 {
+		testCase.Reachability.PrintSummary(true, true, true)
+		framework.Failf("Had %d wrong results in reachability matrix", wrong)
+	}
+	if isVerbose {
+		testCase.Reachability.PrintSummary(true, true, true)
+	}
+	framework.Logf("VALIDATION SUCCESSFUL")
+}
+
+// UpdateNamespaceLabels sets the labels for a namespace
+func UpdateNamespaceLabels(k8s *Scenario, ns string, newNsLabel map[string]string) {
+	err := k8s.setNamespaceLabels(ns, newNsLabel)
+	framework.ExpectNoError(err, "Update namespace %s labels", ns)
+	err = wait.PollImmediate(waitInterval, waitTimeout, func() (done bool, err error) {
+		namespace, err := k8s.getNamespace(ns)
+		if err != nil {
+			return false, err
+		}
+		for key, expected := range newNsLabel {
+			if actual, ok := namespace.Labels[key]; !ok || (expected != actual) {
+				return false, nil
+			}
+		}
+		return true, nil
+	})
+	framework.ExpectNoError(err, "Unable to wait for ns %s to update labels", ns)
+}
+
+// AddPodLabels adds new labels to a deployment's template
+func AddPodLabels(k8s *Scenario, pod *Pod, newPodLabels map[string]string) {
+	kubePod, err := k8s.ClientSet.CoreV1().Pods(pod.Namespace).Get(context.TODO(), pod.Name, metav1.GetOptions{})
+	framework.ExpectNoError(err, "Unable to get pod %s/%s", pod.Namespace, pod.Name)
+	for key, val := range newPodLabels {
+		kubePod.Labels[key] = val
```

Comment:
can we assume that the Labels map was always initialized at this point? or can it be nil?

Reply:
good call, adding a safety check 👍

Reply:
just checked above looks like this is added now

Reply:
checked above, looks like its there now